contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -3328,25 +3346,40 @@ SELECT
 WHERE FALSE;
 GRANT SELECT ON sys.credentials TO PUBLIC;
 
-CREATE VIEW sys.sql_logins AS
-SELECT
-    CAST(NULL as sys.sysname) AS name,
-    CAST(NULL as INT) AS principal_id,
-    CAST(NULL as sys.VARBINARY(85)) AS sid,
-    CAST(NULL as sys.BPCHAR(1)) AS type,
-    CAST(NULL as sys.nvarchar(60)) AS type_desc,
-    CAST(NULL as INT) AS is_disabled,
-    CAST(NULL as sys.DATETIME) AS create_date,
-    CAST(NULL as sys.DATETIME) AS modify_date,
-    CAST(NULL as sys.sysname) AS default_database_name,
-    CAST(NULL as sys.sysname) AS default_language_name,
-    CAST(NULL as INT) AS credential_id,
-    CAST(NULL as INT) AS owning_principal_id,
-    CAST(0 as sys.BIT) AS is_fixed_role,
-    CAST(0 as sys.BIT) AS is_policy_checked,
-    CAST(0 as sys.BIT) AS is_expiration_checked,
-    CAST(NULL as sys.varbinary(256)) AS password_hash
-WHERE FALSE;
+CREATE OR REPLACE VIEW sys.sql_logins AS 
+WITH super_user AS (SELECT pg_get_userbyid(datdba) COLLATE sys.database_default AS super_user FROM pg_database WHERE datname = CURRENT_DATABASE()) 
+SELECT 
+CAST(Base.name AS sys.SYSNAME) AS name,
+CAST(Base.principal_id AS INT) AS principal_id,
+CAST(Base.sid AS sys.varbinary(85)) AS sid,
+CAST('S' AS sys.BPCHAR(1)) AS type,
+CAST('SQL_LOGIN' AS sys.NVARCHAR(60)) AS type_desc,
+CAST(Base.is_disabled AS INT) AS is_disabled,
+CAST(Base.create_date AS SYS.DATETIME) AS create_date,
+CAST(Base.modify_date AS SYS.DATETIME) AS modify_date,
+CAST(Base.default_database_name AS SYS.SYSNAME) AS default_database_name,
+CAST(Base.default_language_name AS SYS.SYSNAME) AS default_language_name,
+CAST(Base.credential_id AS INT) AS credential_id,
+CAST(
+  CASE
+    WHEN Base.name = (SELECT super_user FROM super_user) THEN 0
+    ELSE 1
+  END
+AS sys.BIT) AS is_policy_checked,
+CAST(0 AS sys.BIT) AS is_expiration_checked,
+CAST(
+  CASE
+    WHEN (sys.suser_name() = (SELECT super_user FROM super_user) OR pg_has_role(sys.suser_id(), 'sysadmin'::TEXT, 'MEMBER')) THEN Auth.rolpassword 
+    ELSE NULL
+  END
+AS sys.varbinary(256)) AS password_hash 
+FROM sys.server_principals AS Base 
```

Comment:
from performance pov I think we should not fetch columns from view `server_principals ` rather it should be fetched from base catalog itself. Let's get @HarshLunagariya's opinion on this.